sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -231,9 +229,7 @@ Azure::Security::KeyVault::Keys::RecoverDeletedKeyOperation KeyClient::ResumeRec
     Azure::Core::Context const& context) const
 {
   Azure::Security::KeyVault::Keys::RecoverDeletedKeyOperation operation(m_pipeline, resumeToken);
-  // Need to fix this to use context directly. See:
-  // https://github.com/Azure/azure-sdk-for-cpp/issues/2091
-  operation.Poll(context.GetApplicationContext());
+  operation.Poll(const_cast<Core::Context&>(context));
```

Comment:
This `const_cast` is raising red flags: if the function wants a mutable `Context` argument, it ideally should have a `Context&` parameter instead of `Context const&`. At the very least there should be a comment here explaining why this is `const_cast` is safe.

Reply:
Ah, I fixed it, but forgot to remove the const_cast. Thanks!